test/integration/apiserver/anonymous/anonymous_test.go

```diff
@@ -0,0 +1,332 @@
+
+				resp, err = client.Get(server.ClientConfig.Host + "/healthz")
+				require.NoError(t, err)
+				defer resp.Body.Close()
+				require.Equal(t, 200, resp.StatusCode)
+
+			},
+			additionalFlags: nil,
+		},
+		{
+			desc: "valid config with conditions",
+			featureGates: map[featuregate.Feature]bool{
+				features.StructuredAuthenticationConfiguration: true,
+				features.AnonymousAuthConfigurableEndpoints:    true,
+			},
+			authConfig: `
+apiVersion: apiserver.config.k8s.io/v1beta1
+kind: AuthenticationConfiguration
+anonymous:
+  enabled: true
+  conditions:
+  - path: "/livez"
+`,
+			assertErrFn: func(t *testing.T, errorToCheck error) {
+				require.NoError(t, errorToCheck)
+			},
+			assertFn: func(t *testing.T, server kubeapiserverapptesting.TestServer) {
+				configureRBAC(t, server)
+
+				client := insecureHTTPClient(t)
+
+				resp, err := client.Get(server.ClientConfig.Host + "/api/v1/namespaces/default/pods")
+				require.NoError(t, err)
+				defer resp.Body.Close()
+				require.Equal(t, 401, resp.StatusCode)
+
+				resp, err = client.Get(server.ClientConfig.Host + "/livez")
+				require.NoError(t, err)
+				defer resp.Body.Close()
+				require.Equal(t, 200, resp.StatusCode)
+
+				resp, err = client.Get(server.ClientConfig.Host + "/healthz")
+				require.NoError(t, err)
+				defer resp.Body.Close()
+				require.Equal(t, 401, resp.StatusCode)
+
+			},
+			additionalFlags: nil,
+		},
+	}
+
+	for _, tc := range testCases {
+
+		t.Run(tc.desc, func(t *testing.T) {
+
+			for f, v := range tc.featureGates {
+				featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, f, v)
+
+			}
+
+			flags := []string{"--authorization-mode=RBAC"}
+			flags = append(flags, fmt.Sprintf("--authentication-config=%s", writeTempFile(t, tc.authConfig)))
+			flags = append(flags, tc.additionalFlags...)
+
+			server, err := kubeapiserverapptesting.StartTestServer(
+				t,
+				kubeapiserverapptesting.NewDefaultTestServerOptions(),
+				flags,
+				framework.SharedEtcd(),
+			)
+
+			tc.assertErrFn(t, err)
+
+			if tc.assertFn == nil {
+				return
+			}
+
+			defer server.TearDownFn()
+
+			tc.assertFn(t, server)
+		})
+	}
+
+}
+
+func writeTempFile(t *testing.T, content string) string {
+	t.Helper()
+	file, err := os.CreateTemp("", "oidc-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Cleanup(func() {
+		if err := os.Remove(file.Name()); err != nil {
+			t.Fatal(err)
+		}
+	})
+	if err := os.WriteFile(file.Name(), []byte(content), 0600); err != nil {
+		t.Fatal(err)
+	}
+	return file.Name()
+}
+
+func configureRBAC(t *testing.T, server kubeapiserverapptesting.TestServer) {
+	t.Helper()
+
+	kc := kubernetes.NewForConfigOrDie(server.ClientConfig)
+
+	_, err := kc.RbacV1().Roles(defaultNamespace).Create(context.TODO(), defaultRole, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	_, err = kc.RbacV1().RoleBindings(defaultNamespace).Create(context.TODO(), defaultRoleBinding, metav1.CreateOptions{})
+	require.NoError(t, err)
```

Comment:
we can use context.Background()

many of our tests use TODO and never change that to something else.
the TODO implies it will be changed by someone later.

Reply:
Done.